### PR TITLE
Implemented UNPACK_SEQUENCE instruction

### PR DESCRIPTION
**Pull request by Mdrbhatti**
Implemented UNPACK_SEQUENCE bytecode instruction. Also added a few test cases, which seem to be working for now.
I wasn't able to understand why ```__getitem__``` was returning an ```IPy_Object```, enclosed in brackets. Any feedback would be appreciated!

### CODE REVIEW

**Comment by jvilk** [comment]
I made a number of comments that should help. I think I've helped you with your issue; using a `Py_Slice` with `__getitem__` will return a `Py_List`.

Once you've addressed the comments, this should be ready to merge!

**Comment by Mdrbhatti** [comment]
Thanks a lot for your quick feedback!

I originally implemented it, like you described. But I wasn't able to do this, I would always end up pushing an undefined object on the stack. 
e.g.
For the statement: ```x,y,z = 1,2,3```
I would get (1,2,3) as the result from ```f.pop()```.
I was unsuccessful in all my attempts to extract individual values from this object. I tried ```f.push(val.__getitem__(new Py_Int(x)));``` but it results in an undefined object being pushed to the stack.
Using ```Py_Slice``` with ```__getitem__``` would return me the desired value, but it would be enclosed in round brackets. e.g. (3), instead of 3.
The only way I could push the desired values to the stack, and correctly, was when i converted it into a string and stripped away the brackets.

**Comment by jvilk** [comment]
From what I understand, `i` is a *count* of arguments to unpack. You need to subtract 1 from it, and include 0 in your loop (e.g. `x >= 0`).

**Comment by jvilk** [comment]
Also, `(3)` is what a 1-element `Py_Tuple` prints to the console. e.g. if `foo` is a `Py_Tuple`, `console.log(foo);` will call `foo.toString()` and print that value.

We define `Py_Tuple.toString()` here:

https://github.com/plasma-umass/Ninia/blob/master/src/collections.ts#L229

**Comment by Mdrbhatti** [comment]
I was having trouble understanding what was going on, and I can't believe I did not notice that earlier.
I have committed the recommended changes. Thanks a lot.

File: src/optable.ts [hunk 1]
```diff
@@ -465,6 +465,16 @@ optable[opcodes.STORE_NAME] = function(f: Py_FrameObject) {
     f.locals[name.toString()] = val;
 }
 
+optable[opcodes.UNPACK_SEQUENCE] = function(f: Py_FrameObject) {
+    var val = f.pop();
```
Comment (jvilk) [inline src/optable.ts]: Can you add an `assert(val.__getitem__ !== undefined, "Expected a list or tuple type.");`?

From the [CPython source](https://github.com/python/cpython/blob/master/Python/ceval.c#L2096), it looks like this opcode can also receive an Iterable (which only defines the `__iter__` method). I'd like for us to throw an assertion if we encounter this in a test.

**Comment by jvilk** [comment]
Also, as a note for the future: It looks like this opcode operates on

* Items of *exactly* `Py_List` (so we can actually reach into its guts without worrying about user-defined list types)
* Items of *exactly* `Py_Tuple` (ditto)
* Arbitrary objects that implement `__iter__`

So we can make this opcode a lot more efficient if needed.

**Comment by Mdrbhatti** [comment]
I have added the assertion.

File: src/optable.ts [hunk 2]
```diff
@@ -465,6 +465,19 @@ optable[opcodes.STORE_NAME] = function(f: Py_FrameObject) {
...
+    if(val === undefined) {
```
Comment (jvilk) [inline src/optable.ts]: The assertion should check if the method `val.__getitem__` is undefined, not `val` itself. If that item is missing, then we are not dealing with a list type. We should never have an undefined value on the stack. :)

Comment (Mdrbhatti) [inline src/optable.ts]: Oh, that makes a lot of sense!

**Comment by jvilk** [comment]
Thanks!

**Pull request merged**